packages/cli/src/commands/export-dynamic-plugin/backend.ts

```diff
@@ -0,0 +1,320 @@
+  }
+
+  const filter = {
+    include: mergeWithOutput,
+    exclude: mergeWithOutput.length !== 0 ? undefined : /.*/,
+  };
+  const moveToPeerDependencies: (string | RegExp)[] = [/@backstage\//];
+
+  if (opts.sharedPackage !== undefined) {
+    for (const sharedPkg of opts.sharedPackage as string[]) {
+      moveToPeerDependencies.push(sharedPkg);
+    }
+  }
+
+  const rollupConfigs = await makeRollupConfigs({
+    outputs,
+    minify: Boolean(opts.minify),
+    useApiExtractor: false,
+  });
+
+  if (rollupConfigs.length === 0) {
+    throw new Error('Rollup config is missing');
+  }
+
+  const dependenciesToAdd: {
+    [key: string]: string;
+  } = {};
+
+  const rollupConfig = rollupConfigs[0];
+  rollupConfig.plugins?.push(
+    embedModules({
+      filter: filter,
+      addDependency(embeddedModule, dependencyName, dependencyVersion) {
+        const existingVersion = dependenciesToAdd[dependencyName];
+        if (existingVersion === undefined) {
+          dependenciesToAdd[dependencyName] = dependencyVersion;
+          return;
+        }
+        if (existingVersion !== dependencyVersion) {
+          throw new Error(
+            `several versions ('${existingVersion}', '${dependencyVersion}') of the same transitive dependency ('${dependencyName}') for embedded module ('${embeddedModule}')`,
+          );
+        }
+      },
+    }),
+  );
+
+  if (Array.isArray(rollupConfig.output)) {
+    rollupConfig.output.forEach(output => {
+      if (output.format === 'commonjs') {
+        output.interop = 'default';
+      }
+    });
+  }
+
+  await fs.remove(paths.resolveTarget('dist'));
+
+  try {
+    const bundle = await rollup(rollupConfig);
+    if (rollupConfig.output) {
+      for (const output of [rollupConfig.output].flat()) {
+        await bundle.generate(output);
+        await bundle.write(output);
+      }
+    }
+  } catch (error) {
+    throw new Error(formatErrorMessage(error));
+  }
+
+  if (opts.clean) {
+    await fs.remove(target);
+  }
+
+  const monoRepoPackages = await getPackages(paths.targetDir);
+  await productionPack({
+    packageDir: '',
+    targetDir: target,
+    customizeManifest: (pkgToCustomize: BackstagePackageJson) => {
+      function test(str: string, expr: string | RegExp): boolean {
+        if (typeof expr === 'string') {
+          return str === expr;
+        }
+        return expr.test(str);
+      }
+
+      pkgToCustomize.name = `${pkgToCustomize.name}-dynamic`;
+      (pkgToCustomize as any).bundleDependencies = true;
+      pkgToCustomize.scripts = undefined;
+      pkgToCustomize.types = undefined;
+
+      pkgToCustomize.files = pkgToCustomize.files?.filter(
+        f => !f.startsWith('dist-dynamic/'),
+      );
+
+      for (const dep in dependenciesToAdd) {
+        if (!Object.prototype.hasOwnProperty.call(dependenciesToAdd, dep)) {
+          continue;
+        }
+        if (pkgToCustomize.dependencies === undefined) {
+          pkgToCustomize.dependencies = {};
+        }
```

Comment:
Why is this always reset to empty object? For each item in `dependenciesToAdd` this potentially resets the dependencies. Shouldn't this be outside the for loop?

Also, can't you use:
```suggestion
        pkgToCustomize.dependencies ||= {};
```

Reply:
It is reset to empty object only if it is undefined. So it will be only once afaict.
I do it in the loop so that if `dependenciesToAdd` is empty, we don't change `pkgToCustomize.dependencies` at all, because we won't need to add any new dependency in this (possibly undefined) list.

Reply:
> Also, can't you use:

Fixed. Thanks for the suggestion.